apps/emqx_durable_storage/include/emqx_ds.hrl

```diff
@@ -16,4 +16,28 @@
 -ifndef(EMQX_DS_HRL).
 -define(EMQX_DS_HRL, true).
 
+-record(dsbatch, {
+    operations :: [emqx_ds:operation()],
+    preconditions = [] :: [emqx_ds:precondition()]
+}).
+
+-record(message_matcher, {
+    %% Fields identifying the message:
+    %% Client identifier
+    from :: binary(),
+    %% Topic that the message is published to
+    topic :: emqx_types:topic(),
+    %% Timestamp (Unit: millisecond)
+    timestamp :: integer(),
+
+    %% Fields the message is matched against:
+    %% Message Payload
+    payload,
+    %% Message headers
+    headers = #{} :: emqx_types:headers(),
```

Comment:
Nit: if we don't support matching of certain fields (like headers), maybe they should not be here to avoid making a false promise to the API consumer.

Reply:
True. But this one is something that could be demanded in the near future and trivial to implement (e.g. with _subset of_ semantics).